torch/_inductor/codegen/wrapper.py

```diff
@@ -599,6 +606,7 @@ def is_expr(x):
             for dim, shape in enumerate(shapes):
                 shape = V.graph.sizevars.simplify(shape)
                 if shape in needed:
+                    self.defined_symbols.add(shape)
```

Comment:
This seems wrong.  `shape` might be `s0+s1+1`.

Perhaps you want `.free_symbols`?

Reply:
```shape``` is guarded by ```needed```, which is ```needed = V.graph.sizevars.free_symbols()``` at line 576 above, so we won't generate bad code for non-symbol expressions such as ```s0 + s1```.

Reply:
I am worried we will miss the symbols though.

`s0+s1 in {s0}` will return false and defined symbols will be wrong.

Reply:
@jansel The issue is addressed in https://github.com/pytorch/pytorch/pull/110411, where I removed the source of generating such bad code. There was some historical reason why we chose the previous approach. I think the changes in #110411 make the implementation much simpler and less error-prone.